### ISSUE TITLE
stencils-check not providing results when access specification is required

### ISSUE THREAD

**Issue by GuiltyDolphin** [camfort/camfort#94]
E..g, with the following program (`example.f90`):

```fortran
program example
  integer :: i
  real, dimension(10) :: a
  do i = 1, 10
    != stencil pointed(dim=1) :: a
    x = a(i)
  end do
end program
```

Running `camfort stencils-check example.f90` produces:
```
Checking stencil specs for 'example.f90'

```

But we should be validating the existing stencil (in this case, a stencil of `!= access readOnce, (pointed(dim=1)) :: a` should be inferred. Using the correct specification also results in no validation).

**Comment by dorchard** [comment]
Good spot. We should be complaining that a `stencil` spec has been given rather than an `access` spec.

**Comment by dorchard** [comment]
Fixed by 13fa126778ebf79774cec994a4bc387000d73987
